### PR TITLE
Generate a final preview for the notulen taking into account public behandelings

### PR DESCRIPTION
**Pull request by lagartoverde**

### CODE REVIEW

**Comment by lagartoverde** [comment]
Requires https://github.com/lblod/notulen-prepublish-service/pull/81

**Review by benjay10 — APPROVED** [review]
Seems to work fine!

Maybe a suggestion: should we notify the user on the preview modal that the content of the notulen are not public? I'm just thinking about a colored tag on the side, with some label, to make it clearer.

**Comment by lagartoverde** [comment]
That seems like a good idea, but it would require design, maybe post it on the chat and see what the others think :)

File: app/controllers/meetings/publish/notulen.js [hunk 1]
```diff
@@ -146,11 +146,26 @@ export default class MeetingsPublishNotulenController extends Controller {
...
       headers: { 'Content-Type': 'application/vnd.api+json' },
       body: JSON.stringify({
-        'public-behandeling-uris': this.publicBehandelingUris,
+        data: {
+          type: 'meeting-notes-previews',
+          relationships: {
+            publicTreatments: this.publicBehandelingUris.map((uri) => ({
+              data: {
+                id: uri,
+                type: 'treatment',
```
Comment (nvdk) [inline app/controllers/meetings/publish/notulen.js:159]: I think types need to be plural in json:api

File: app/controllers/meetings/publish/notulen.js [hunk 2]
```diff
@@ -146,11 +146,26 @@ export default class MeetingsPublishNotulenController extends Controller {
...
+              },
+            })),
+            meeting: {
+              data: {
+                id: meetingId,
```
Comment (nvdk) [inline app/controllers/meetings/publish/notulen.js:164]: type is missing here

**Comment by nvdk** [comment]
so... I was debugging some issues and that turned into quite a large refactor :). this now needs https://github.com/lblod/notulen-prepublish-service/tree/enhancement/aync-notulen-preview-generation to work. @lagartoverde I'd appreciate it if you could test it a bit before we merge

**Review by lagartoverde — COMMENTED** [review]
I cannot approve this as is my own PR but code looks good, works good and I'm happy it got updated, I was thinking about doing it, but didn't have enough knowledge of the whole thing to be brave to refactor it. I have tested extensively and didn't get any errors but you never know with the publication

**Pull request merged**